mewarpx/mewarpx/diags_store/checkpoint_diagnostic.py

```diff
@@ -36,49 +36,39 @@ def __init__(self, diag_steps,
         self.checkpoint_steps = diag_steps
         self.name = name
         self.clear_old_checkpoints = clear_old_checkpoints
+        self.num_to_keep = num_to_keep
         self.flux_diag = None
 
-        if kwargs.pop('diag_step_offset', 1) != 1:
-            raise ValueError(
-                "diag_step_offset in CheckPointDiagnostic must be 1.")
         super(CheckPointDiagnostic, self).__init__(
-            diag_steps=diag_steps, diag_step_offset=1, **kwargs)
+            diag_steps=diag_steps, **kwargs)
 
         self.write_dir = self.DIAG_DIR
         self.add_checkpoint()
 
-        # note that WarpX diagnostics (including checkpoints) are output
-        # after the "afterstep" functions, therefore we install this function
-        # before steps and add 1 to the diag_steps above
         if self.checkpoint_steps != mwxrun.simulation.max_steps:
-            callbacks.installbeforestep(self.regular_checkpoint_manager)
+            callbacks.installafterdiagnostics(self.checkpoint_manager)
```

Comment:
Is it still necessary to check that checkpoint_steps != max_steps? It seems like before that was necessary because this would only run the step after the simulation completed, but now I think it's still fine?

Reply:
I guess it isn't necessary but having it there avoids installing an extra callback that will check the timestep every step but never do anything useful. So I prefer leaving it in, but happy to take it out if you prefer that.